perma_web/perma/models.py

```diff
@@ -1459,30 +1459,38 @@ def wr_iframe_prefix(self, wr_username):
         return "{}/{}/{}/".format(settings.PLAYBACK_HOST, wr_username, self.wr_collection_slug)
 
     def init_replay_for_user(self, request):
-        wr_username, wr_session_cookie, is_new_session = get_wr_session(request)
-        if is_new_session or not get_wr_uploaded(request, self.wr_collection_slug):
+        is_private = self.is_private
+        wr_username, wr_session_cookie, is_new_session = get_wr_session(request, is_private, self.wr_collection_slug)
+        if is_new_session:
             try:
-                self.upload_to_wr(wr_username, wr_session_cookie)
-                set_wr_uploaded(request, self.wr_collection_slug)
+                self.upload_to_wr(wr_username, wr_session_cookie, is_private)
             except WebrecorderException:
                 clear_wr_session(request)
                 raise
         return wr_username
 
-    def upload_to_wr(self, wr_username, wr_session_cookie):
+    def upload_to_wr(self, wr_username, wr_session_cookie, is_private):
```

Comment:
Here `self` is the link: you can just use the `self.is_private` attribute, rather than passing `is_private` as an argument to the method. Both for brevity, and to avoid any possibility of bugs where a person might call `link.upload_to_wr` on a private link but then pass in `False` for `is_private`.

Reply:
Yep, updated this in the refactoring.